tests/Backend/Snowflake/TimeTravelTest.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace Keboola\Test\Backend\Snowflake;
+
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\TableExporter;
+use Keboola\Test\StorageApiTestCase;
+
+class TimeTravelTest extends StorageApiTestCase
+{
+    private $downloadPath;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->_initEmptyTestBuckets();
+        $this->downloadPath = __DIR__ . '/../../_tmp/';
+    }
+    
+    public function testCreateTableFromTimestamp()
+    {
+        $importFile = new CsvFile(__DIR__ . '/../../_data/languages.csv');
+
+        $sourceTable = 'languages_' . date('Ymd_His');
+        
+        $sourceTableId = $this->_client->createTable(
+            $this->getTestBucketId(),
+            $sourceTable,
+            $importFile
+        );
+        $originalTable = $this->_client->getTable($sourceTableId);
+        // the timestamp must be at least 1 sec > creation time
+        sleep(5);
+        $timestamp = date(DATE_ATOM);
+        sleep(25);
+
+        $this->_client->writeTable($sourceTableId, $importFile, ['incremental' => true]);
+
+        $updatedTable = $this->_client->getTable($sourceTableId);
+        $newTableName = "new-table-name_" . date('Ymd_His', strtotime($timestamp));
+
+        $replicaTableId = $this->_client->createTableFromSourceTableAtTimestamp(
+            $this->getTestBucketId(self::STAGE_OUT),
+            $sourceTableId,
+            $timestamp,
+            $newTableName
+        );
+
+        $replicaTable = $this->_client->getTable($replicaTableId);
+
+        $this->assertEquals($newTableName, $replicaTable['name']);
+
+        // Pending fix of timetravel metadata issue case 00022189
+        //
+        // $this->assertEquals($updatedTable['rowsCount'], $replicaTable['rowsCount'] * 2);
+        // $this->assertEquals($originalTable['rowsCount'], $replicaTable['rowsCount']);
+
+        // test data export
+        $exporter = new TableExporter($this->_client);
+        $downloadFile = $this->downloadPath . 'timeTravelOutput.csv';
+        $exporter->exportTable($replicaTableId, $downloadFile, []);
+        $this->assertArrayEqualsSorted(
+            Client::parseCsv(file_get_contents($importFile)),
+            Client::parseCsv(file_get_contents($downloadFile)),
+            'id'
+        );
+    }
+
+    public function testCreateTableFromTimestampOfAlteredTable()
+    {
+        $importFile = new CsvFile(__DIR__ . '/../../_data/languages.csv');
+
+        $sourceTable = 'languages_' . date('Ymd_His');
+
+        $sourceTableId = $this->_client->createTable(
+            $this->getTestBucketId(),
+            $sourceTable,
+            $importFile
+        );
+        $originalTable = $this->_client->getTable($sourceTableId);
```

Comment:
You are creating a few unused variables here, please check it in PHPStorm these are highlighted.